asteroid/data/musdb18_dataset.py

```diff
@@ -0,0 +1,213 @@
+        split (str, optional): Dataset subfolder, defaults to `train`.
+        subset (:obj:`list` of :obj:`str`, optional): Selects a specific of
+            list of tracks to be loaded, defaults to `None` (loads all tracks).
+        segment (float, optional): Duration of segments in seconds,
+            defaults to ``None`` which loads the full-length audio tracks.
+        samples_per_track (int, optional):
+            Number of samples yielded from each track, can be used to increase 
+            dataset size, defaults to `1`.
+        random_segments (boolean, optional): Enables random offset for track segments.
+        random_track_mix boolean: enables mixing of random sources from
+            different tracks to assemble mix.
+        source_augmentations (:obj:`list` of `obj`:`callable`):
+            list of augmentation function names,
+            defaults to no-op augmentations (input = output)
+        sample_rate (int, optional): Samplerate of files in dataset.
+        tracks (:obj:`list` of :obj:`Dict`): List of track metadata
+
+    """
+    def __init__(self,
+                 root,
+                 sources=['vocals', 'bass', 'drums', 'other'],
+                 suffix='.wav',
+                 split='train',
+                 subset=None,
+                 segment=None,
+                 samples_per_track=1,
+                 random_segments=False,
+                 random_track_mix=False,
+                 source_augmentations=lambda audio: audio,
+                 sample_rate=44100):
+
+        self.root = Path(root).expanduser()
+        self.split = split
+        self.sample_rate = sample_rate
+        self.segment = segment
+        self.random_track_mix = random_track_mix
+        self.random_segments = random_segments
+        self.source_augmentations = source_augmentations
+        self.sources = sources
+        self.suffix = suffix
+        self.subset = subset
+        self.samples_per_track = samples_per_track
+        self.tracks = list(self.get_tracks())
+        if not self.tracks:
+            raise RuntimeError("No tracks found.")
+
+    def __getitem__(self, index):
+        # assemble the mixture of target and interferers
+        audio_sources = {}
+
+        # get track_id
+        track_id = index // self.samples_per_track
+        if self.random_segments:
+            start = random.uniform(
+                0, self.tracks[track_id]['min_duration'] - self.segment
+            )
+        else:
+            start = 0
+
+        # load sources
+        for source in self.sources:
+            # optionally select a random track for each source
+            if self.random_track_mix:
+                # load a different track
+                track_id = random.choice(range(len(self.tracks)))
+                if self.random_segments:
+                    start = random.uniform(
+                        0, self.tracks[track_id]['min_duration'] - self.segment
+                    )
```

Comment:
This makes sense to me, at least, when we don't mix track together, we keep things aligned